src/components/hooks/useMeetingDestination.js

```diff
@@ -26,39 +26,42 @@ export default function useMeetingDestination(meetingDestination) {
     localVideo: null,
     remoteAudio: null,
     remoteVideo: null,
-  };
-  const endedMeeting = {
-    ...emptyMeeting,
-    ID: null,
+    state: null,
   };
 
   const [meeting, setMeeting] = useState(emptyMeeting);
   const {meetingsAdapter} = useContext(AdapterContext);
 
   useEffect(() => {
-    // React won't recognize the meeting attributes have been updated
-    // since the state is the meeting object itself
-    // We need to create a new meeting object trigger the state change
-    const onMeeting = (newMeeting) => {
-      setMeeting({...newMeeting});
-    };
-    const onError = (error) => {
-      setMeeting(emptyMeeting);
-      // eslint-disable-next-line no-console
-      console.log(error);
-    };
-    const onComplete = () => {
-      setMeeting(endedMeeting);
-    };
+    let cleanup;
+
+    if (!meetingDestination) {
+      setMeeting({...emptyMeeting});
+      cleanup = undefined;
+    } else {
+      const onMeeting = (newMeeting) => {
+        // React won't recognize the meeting attributes have been updated
+        // since the state is the meeting object itself
+        // We need to create a new meeting object trigger the state change
+        setMeeting({...newMeeting});
+      };
+      const onError = (error) => {
+        setMeeting({...emptyMeeting, error});
+        // eslint-disable-next-line no-console
+        console.log(error);
+      };
+
+      const subscription = meetingsAdapter
+        .createMeeting(meetingDestination)
+        .pipe(concatMap(({ID}) => meetingsAdapter.getMeeting(ID)))
+        .subscribe(onMeeting, onError);
 
-    const subscription = meetingsAdapter
-      .createMeeting(meetingDestination)
-      .pipe(concatMap(({ID}) => meetingsAdapter.getMeeting(ID)))
-      .subscribe(onMeeting, onError, onComplete);
+      cleanup = () => {
+        subscription.unsubscribe();
+      };
+    }
 
-    return () => {
-      subscription.unsubscribe();
-    };
+    return cleanup;
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Let's fix this linter issue instead of disabling. We should re-run the hook if the adapter changes